src/Internal/Payment/State/VerifyState.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Class VerifyState
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment\State;
+
+use WC_Payments_API_Payment_Intention;
+use WCPay\Constants\Payment_Intent_Status;
+use WCPay\Core\Exceptions\Server\Request\Extend_Request_Exception;
+use WCPay\Core\Exceptions\Server\Request\Immutable_Parameter_Exception;
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Internal\Service\OrderService;
+use WCPay\Internal\Service\PaymentRequestService;
+
+/**
+ * This state is used to create payment intent and verify next actions based on the intent state.
+ */
+class VerifyState extends AbstractPaymentState {
```

Comment:
Let's make this `VerifiedState`, as verification is already done.

Reply:
Good point. Refactored here: https://github.com/Automattic/woocommerce-payments/pull/7471/commits/585514936cd51dbe088123bba8615412d3a3272b